### DISCUSSION TITLE
I made a Phlex/Sinatra adapter

### DISCUSSION THREAD

**Discussion by benpickles** [yippee-fun/phlex#565] [Show and tell]
Hi everyone 👋🏻

Obviously Phlex already works with Sinatra (and everything else) but its normal usage leaves you without access to Sinatra's standard helper methods - most importantly for me the `#url` method - so I made [phlex-sinatra](https://github.com/benpickles/phlex-sinatra).

There are a few more things I'd like to add but it's ready to use right now if lack of the `#url` helper has so far been holding you back from using Phlex with Sinatra.

I'm also interested in streaming integration - I haven't looked it at all but it's on my mind.

**Comment by joeldrapper** [comment]
This looks great @benpickles.

**Reply by benpickles** [reply]
I wrote a little bit about the [phlex-sinatra and why I created it](https://www.benpickles.com/articles/92-phlex-sinatra) and also about [the (micro)site I was working on at the time](https://www.benpickles.com/articles/91-my-wordle-results) – I apologise for the quote that Phlex "uglifies by default" but it's true!

**Reply by joeldrapper** [reply]
😆 Yes, it is true. I considered adding a configuration for pretty output but couldn't find any reason to do that. If performance is important, you'll almost definitely want uglified output. If you want pretty output, you can just run the uglified output through a prettifier and don't worry about performance.
